client/state/terms/actions.js

```diff
@@ -35,7 +37,15 @@ export function addTerm( siteId, taxonomy, term ) {
 		} ) );
 
 		return wpcom.site( siteId ).taxonomy( taxonomy ).term().add( term ).then(
-			( data ) => dispatch( receiveTerm( siteId, taxonomy, omit( data, '_headers' ) ) ),
+			( data ) => {
+				const newTerm = omit( data, '_headers' );
+				dispatch( receiveTerm( siteId, taxonomy, newTerm ) );
+
+				// if a postId is set dispatch action to select term
+				if ( postId !== null ) {
```

Comment:
Should this pass when `postId` is undefined (i.e. when the last argument is omitted)? It appears it will currently.